magento/magento2
# Update lessjs to v3.13.1

### Description (*)


Update old version to recently newer
2.3.1 to 3.13.1
Current magento used  lessjs v2.3.1.
Official version is https://cdn.jsdelivr.net/npm/less@3.13
https://www.jsdelivr.com/package/npm/less
For maintain compatible we can stay at v3 for a while!. Current official less released v4.1.0


Official less website http://lesscss.org/

This file allow to browser compile less file to css on-flies. Allow developer can check styles from browser without need run deploy ! Note this feature not work in production mode

Idealy lessjs the version should be same as lessphp compile version


### Related Pull Requests


### Fixed Issues (if relevant)

- Fixes https://github.com/magento/magento2/issues/32845

### Manual testing scenarios (*)

Node should have version >=12 <14
Current grunt not work properly in node v14+

Test case 1
1. Enable developer mode
2. Enable Client side compilation in Advanced Configuration
 (Stores > Configuration > Advanced >  Developer
Frontend Development Workflow => Workflow type : Client side less compilation)
3. Save and clear cache config admin
4. Go to frontend and verify page render properly without break or glitch

Test case 2
1. Change package.json.sample => package.json
2. Install package by run command
npm install
3. from command-line run 
node_modules/.bin/lessc --version and check properly version less compare with (package.json)

Test case 3
Run grunt workflow without problem
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/css-topics/css_debug.html

https://devdocs.magento.com/guides/v2.4/test/js/jasmine.html
Noted before run grunt spec must deploy static contents first

### Questions or comments


CC: @Leland @krzksz  @ptylek what do you guys think on this ?

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#31824: Update lessjs to v3.13.1
